libs/vectorise/include/vectorise/fixed_point/fixed_point.hpp

```diff
@@ -1960,6 +1964,130 @@ constexpr FixedPoint<I, F> &FixedPoint<I, F>::operator*=(FixedPoint<I, F> const
   return *this;
 }
 
+/**
+ * Multiplication assignment operator, multiply given object to self
+ * @param the given FixedPoint object to multiply against
+ * @return the product of the two FixedPoint objects
+ */
+template <>
+constexpr FixedPoint<64, 64> &FixedPoint<64, 64>::operator*=(FixedPoint<64, 64> const &n)
+{
+  if (IsNaN(*this) || IsNaN(n))
+  {
+    fp_state |= STATE_NAN;
+    *this = NaN;
+    return *this;
+  }
+
+  bool this_pos_inf = IsPosInfinity(*this);
+  bool this_neg_inf = IsNegInfinity(*this);
+  bool n_pos_inf    = IsPosInfinity(n);
+  bool n_neg_inf    = IsNegInfinity(n);
+  bool this_zero    = *this == _0;
+  bool n_zero       = n == _0;
+  bool this_pos     = *this > _0;
+  bool this_neg     = *this < _0;
+  bool n_neg        = n < _0;
+
+  // Multiplying  +∞/-∞ with 0 gives NaN
+  bool nan = (n_zero && (this_pos_inf || this_neg_inf)) || (this_zero && (n_pos_inf || n_neg_inf));
+  // Multiplying +∞ with any positive number (incl +∞) or -∞ with a negative number (incl -∞), gives
+  // +∞
+  bool pos_inf = (this_pos_inf && n > _0) || (n_pos_inf && this_pos) || (this_neg_inf && n < _0) ||
+                 (n_neg_inf && this_neg);
+  // Multiplying +∞ with any negative number (incl -∞) or -∞ with a positive number (incl. +∞) gives
+  // -∞
+  bool neg_inf = (this_pos_inf && n < _0) || (n_pos_inf && this_neg) || (this_neg_inf && n > _0) ||
+                 (n_neg_inf && this_pos);
+
+  if (nan)
+  {
+    fp_state |= STATE_NAN;
+    *this = NaN;
+    return *this;
+  }
+  if (pos_inf)
+  {
+    fp_state |= STATE_INFINITY;
+    *this = POSITIVE_INFINITY;
+    return *this;
+  }
+  if (neg_inf)
+  {
+    fp_state |= STATE_INFINITY;
+    *this = NEGATIVE_INFINITY;
+    return *this;
+  }
+
+  // We duplicate multiplication as done in the Int<> class
+  bool sign = true;
+  if (this_neg)
+  {
+    data_ = -data_;
+  }
+  Type other{n.Data()};
+  if (n_neg)
+  {
+    other = -other;
+  }
+  sign = !(this_neg ^ n_neg);  // NOLINT
```

Comment:
Try
```
sign = !static_cast<bool>(this_neg ^ n_neg);
```
to make static analysis happy - it wasn't complaining that you're converting to `bool`, just that the conversion was implicit instead of explicit.

Reply:
Alternatively you could have tried comparing to zero rather than to `false` instead of casting.